firebase-sessions/src/main/kotlin/com/google/firebase/sessions/ProcessDetails.kt

```diff
@@ -40,23 +46,23 @@ internal class AndroidProcessDetails(context: Context) : ProcessDetails {
    * This is the app's package name unless the app overrides the android:process attribute in the
    * application block of its Android manifest file.
    */
-  private val defaultProcessName: String = context.applicationInfo.processName
+  override val defaultProcessName: String = context.applicationInfo.processName
 
-  /** The name of the current process, or null if it couldn't be found. */
-  private val currentProcessName: String? =
+  /** The name of this process, or null if it couldn't be found. */
+  override val processName: String? =
     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
       Application.getProcessName()
     } else {
       findProcessName(context, Process.myPid())
     }
 
-  override val isDefaultProcess: Boolean = currentProcessName == defaultProcessName
+  override val isDefaultProcess: Boolean = processName == defaultProcessName
 
   override val isForegroundProcess: Boolean
     get() {
       val runningAppProcessInfo = RunningAppProcessInfo()
       ActivityManager.getMyMemoryState(runningAppProcessInfo)
-      return runningAppProcessInfo.importance == RunningAppProcessInfo.IMPORTANCE_FOREGROUND
+      return runningAppProcessInfo.importance <= RunningAppProcessInfo.IMPORTANCE_FOREGROUND_SERVICE
```

Comment:
I think this should go back to `runningAppProcessInfo.importance == RunningAppProcessInfo.IMPORTANCE_FOREGROUND` because `IMPORTANCE_FOREGROUND_SERVICE` is only available after a specific api level, and it'll make our logic complicated to also count services as leaders.

Reply:
the one integration test fails if we don't consider foreground services to be foreground processes https://github.com/firebase/firebase-android-sdk/blob/master/firebase-sessions/test-app/src/androidTest/kotlin/com/google/firebase/testing/sessions/FirebaseSessionsTest.kt

That suggests to me that we may need it. Thoughts?